### ISSUE TITLE
Problem while using cpu relocation in multi_gpu_model

### ISSUE THREAD

**Issue by hsjimwang** [keras-team/keras#13177]
I am new to Keras and now trying to train a model by data parallelism, and luckily I have found the multi_gpu_model function which is already built in Keras module. 

However, after using multi_gpu_model with 4 gpus, it gets much slower than even single gpu. 

And I tried every means until setting the argument "cpu_relocation" as True, then the program finally speeds up by around 2 times on 4 gpus.

Here is my code for data parallelism on multi gpus:
```
if gpus <= 1:)
	print("[INFO] training with 1 GPU..."))
else:)
	print("[INFO] training with {} GPUs...".format(gpus)))
	# make the model parallel)
	parallel_model = multi_gpu_model(model, gpus=gpus, cpu_relocation=True)
```

I am wonder why it failed when setting multi_gpu_model arguments by default. Is it because my model too big? (10756 MiB / 11172 MiB)  

I have searched for lots githiub site about using multi_gpu_model but did not find anyone set "cpu_relocation=True"

Already read through the comments in the official Keras sites but still do not understand.

> cpu_relocation: A boolean value to identify whether to create the model's weights under the scope of the CPU. If the model is not defined under any preceding device scope, you can still rescue it by activating this option.

Could anyone please give me some explanation about this problem? Any advice for using multiple gpu on Keras is welcome!

Thank you in advance!

**Comment by HuangChiEn** [comment]
As the document description: create the model's weights under the scope of the CPU.
Most of the model build in tensorflow by explicit declare the device scope. The multi_gpu_model allow the model deployed on the GPU device, which means the data(model weight, gradient of each layer, batch image) will be move in/out from the cpu to the gpu, and the IO may bound the performance of speed.
"cpu_relocation" allow the model weight bind on the cpu device, which may speed up the update of the weight. (I guess that's why the program finally speeds up by around 2 times on 4 gpus)
